.github/workflows/archive-extension-source-code.yml

```diff
@@ -0,0 +1,70 @@
+name: Archive Extension Build And Source Code
+
+on:
+  push:
+    tags:
+      - coil-extension@*
+
+jobs:
+  save-monorepo-source-code:
+    runs-on: ubuntu-latest
+   
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
+      - name: Step 1 - Use the Upload Artifact GitHub Action to upload the monorepo
+        uses: actions/upload-artifact@v2
```

Comment:
https://github.com/actions/download-artifact
Download artifact is up to version 3, should use that, it might have perf improvements :)